### ISSUE TITLE
Indentation not working

### ISSUE THREAD

**Issue by aik2mlj** [uben0/tree-sitter-typst#43]
First of all, thank you for your work!

I find out that indentation for typst files is correct in vanilla neovim. However, after installing this treesitter from `nvim-treesitter`, the indentation stops working as expect.
E.g.,
```typst
#{
  let n = 0  // if you insert a new line here, the cursor is supposed to be indented. But instead it isn't after installing this treesitter.
}
```

I've tested on both neovim v0.10.2 and v0.11.0-dev.

**Comment by uben0** [comment]
Indeed, with a tree-sitter grammar, there is supposed to be a indentation query file. I lazily postponed the creation of this file...

I can do it for the Helix editor (the editor I use), as this is documented [here](https://docs.helix-editor.com/guides/indent.html) but I don't know how standard it is. Maybe it will work the same with NeoVim.

**Comment by bowentan** [comment]
Hi! An off-topic question that I found an `indent.scm` in the tree-sitter repo but it is missing here. (I checked the lockfile for this repo in tree-sitter repo, they were matched actually.) I think the source codes are maintained here, right? If I want to help debug the indentations for typst, should I fork this or directly tree-sitter? 🤔

BTW, even with that `indent.scm`, indentations are not working as expected. See [this issue](https://github.com/nvim-treesitter/nvim-treesitter/issues/7815) for details.

**Comment by ArttuLai** [comment]
I think I have an OK setup in nvim that covers most of the indentation woes. I added these queries to a `~/.config/nvim/queries/typst/indents.scm` file. These might not cover all the situations, as I’m no expert, but hey.

```
[
  (block)
  (group)
  (tagged
    (content))
  (group
    (content))
  (code
    (content))
  (math)
] @indent.begin

[
  (item
    (text))
  (comment)
  (ERROR)
] @indent.auto

[
  "("
  ")"
  "["
  "]"
  "{"
  "}"
  "$"
] @indent.branch

[
  ")"
  "]"
  "}"
  "$"
] @indent.end
```

## Example

Before:

```typst
#{
  code
  <CR>
}
```

After:

```typst
#{
  code
  <cursor>
}
```

## Example

Before:

```typst
#figure(
  image: "test.png"<CR>
)
```

After:

```typst
#figure(
  image: "test.png"
  <cursor>
)
```

## Example

Before:

```typst
- Item 1
  - Nested item 1<CR>
```

After:

```typst
- Item 1
  - Nested item 1
  <cursor>
```

**Comment by aik2mlj** [comment]
I later simply bypassed this treesitter and reverted to neovim's default indentation engine by
```lua
require'nvim-treesitter.configs'.setup {
  indent = {
    enable = true,
    disable = { "typst" },
  },
}
```

Hope it helps for other people having this issue.
